src/cryptonote/slow-hash.c

```diff
@@ -196,7 +196,7 @@ void cn_slow_hash(const char* input, char* output, int len, int variant, uint32_
   union cn_slow_hash_state state;
   uint8_t text[INIT_SIZE_BYTE];
   uint8_t a[AES_BLOCK_SIZE];
-  uint8_t b[AES_BLOCK_SIZE];
+  uint8_t b[AES_BLOCK_SIZE * 2];
```

Comment:
This was the root of the problem.  Depending on the compiler options, writing to b + AES_BLOCK_SIZE would write into c or a.  One was ok, the other killed the hash.